docs/react-migration-guide.md

```diff
@@ -1,6 +1,208 @@
+  {children}
+</ChannelProvider>
+
+// in a child component:
+useChannel({ channelName: 'your-derived-channel-name' }, (message) => {
+  console.log(message);
+});
+```
+
+Replace code that used `options` in `usePresence` hook:
+
+```jsx
+const { updateStatus } = usePresence(
+  { channelName: 'presence-channel-name', options: { modes: ['PRESENCE'] } },
+  { foo: 'bar' },
+);
+```
+
+With this:
+
+```jsx
+// in a parent component:
+return <ChannelProvider channelName="presence-channel-name" options={{ modes: ['PRESENCE'] }}>
+  {children}
+</ChannelProvider>
+
+// in a child component:
+const { updateStatus } = usePresence({ channelName: 'presence-channel-name' }, { foo: 'bar' });
+```
+
+Additionally, if you were calling `.setOptions()` on a channel instance returned by the `useChannel` hook before, you must remove those calls and instead modify options provided to the `ChannelProvider` component if you want to change channel options during runtime.
+
+Change this:
+
+```jsx
+const { channel } = useChannel({ channelName: 'your-channel-name' }, (message) => {
+  console.log(message);
+});
+channel.setOptions({ params: { rewind: '1' } });
+```
+
+To this:
+
+```jsx
+// in a parent component:
+// change channel options during runtime using state
+const [options, setOptions] = useState({});
+return <ChannelProvider channelName="your-channel-name" options={options}>
+  {children}
+</ChannelProvider>
+
+// in a child component:
+const { channel } = useChannel({ channelName: 'your-channel-name' }, (message) => {
+  console.log(message);
+});
+```
+
+### `usePresence` is split into two hooks
+
+The functionality of the `usePresence` hook has been split into two separate hooks.
+
+The `usePresence` hook can now only be used to enter the presence with optional initial state and update the presence status for the current client. It no longer returns the `presenceData` value and does not accept the `onPresenceUpdated` callback as its third parameter.
+
+To listen for presence updates, a new hook called `usePresenceListener` has been introduced. This hook returns the `presenceData` object previously returned by `usePresence` and accepts an `onPresenceMessageReceived` callback as its second parameter, which is called on new presence messages.
+
+Replace this:
+
+```jsx
+const { presenceData, updateStatus } = usePresence(
+  { channelName: 'presence-channel-name' },
+  { foo: 'bar' },
+  (update) => {
+    console.log(update);
+  },
+);
+```
+
+With this:
+
+```jsx
+const { updateStatus } = usePresence({ channelName: 'presence-channel-name' }, { foo: 'bar' });
+const { presenceData } = usePresenceListener({ channelName: 'presence-channel-name' }, (update) => {
+  console.log(update);
+});
+```
+
+### Renaming `id` to `ablyId`
+
+All instances of the `id` field, which were used to specify the `AblyProvider` component and the underlying Ably client to use, have been renamed to `ablyId`.
```

Comment:
I think it would be nice to make it clear that this was an optional feature, i'd bet the vast majority of users aren't using more than one provider so this section might be a bit confusing to them
```suggestion
All instances of the `id` field, which optionally were used to specify the `AblyProvider` component and the underlying Ably client to use, have been renamed to `ablyId`.
```

Reply:
True, will also change the header to `Renaming optional "id" field to "ablyId"`